### PR TITLE
Update the padding of text under a form item

### LINKED ISSUE #4137: form hint text has confusing padding to next widget
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

the `HintText` and the validation message has more padding to the widget it belongs to, rather than to the next widget. I think this is visually confusing, since the hint is meant for the preceding widget.

a fix would be to reduce the top padding and/or increase the bottom padding of hints/validation messages.

### How to reproduce

open fyne demo

### Screenshots

Screenshot 2023-08-05 at 18 34 37

### Example code

fyne demo

### Fyne version

2.3.5

### Go compiler version

1.20.6

### Operating system and version

macOS 13.5

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by andydotxyz**
Fixes #4137

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: widget/form.go [hunk 1]
```diff
@@ -371,3 +370,21 @@ func NewForm(items ...*FormItem) *Form {
...
+
+	objs[1].Move(fyne.NewPos(theme.InnerPadding(), itemHeight+theme.InnerPadding()/2))
+	objs[1].Resize(fyne.NewSize(size.Width, objs[1].MinSize().Width))
+}
+
+func (f formItemLayout) MinSize(objs []fyne.CanvasObject) fyne.Size {
+	min0 := objs[0].MinSize()
+	min1 := objs[1].MinSize()
+
+	minWidth := fyne.Max(min0.Width, min1.Width)
```
Comment (Jacalz) [inline widget/form.go:388]: Should this not be called "maxWidth"?

File: widget/form.go [hunk 2]
```diff
@@ -371,3 +370,21 @@ func NewForm(items ...*FormItem) *Form {
...
+	return fyne.NewSize(minWidth, min0.Height+min1.Height).AddWidthHeight(0, theme.InnerPadding())
```
Comment (Jacalz) [inline widget/form.go:389]: No need for the extra add here. You can move it into the size creation.
```suggestion
	return fyne.NewSize(minWidth, min0.Height+min1.Height+theme.InnerPadding())
```

**Review by Jacalz — CHANGES_REQUESTED** [review]
I pushed a commit to fix the tests. Just one inline comment about a variable name and a minor cleanup while you're at it.

File: widget/form.go (hunk 1 above)
Comment (andydotxyz) [reply widget/form.go:388]: No, the max of to minimums is still a minimum

**Comment by andydotxyz** [comment]
> I pushed a commit to fix the tests. Just one inline comment about a variable name and a minor cleanup while you're at it.

Thanks - between replying and accepting the simplification I think we're good?

**Review by Jacalz — APPROVED** [review]
Sorry, I thought I had approved

**Comment by pseidemann** [comment]
thanks 👍

**Pull request merged**